### ISSUE TITLE
Support for devfile API events.preStop

### ISSUE THREAD

**Issue by AObuchow** [devfile/devworkspace-operator#1139]
### Description
Currently, `events.preStop` from the devfile API is [not supported by DWO](https://github.com/devfile/devworkspace-operator/blob/main/docs/unsupported-devfile-api.adoc). 

Support for multiple `preStop` events should be added to DWO. 

When this is implemented, it should not conflict with the [async storage's preStop event hook](https://github.com/devfile/devworkspace-operator/blob/b4fc4ea16f8bd294d3e7b3a3a76db17683450169/pkg/provision/storage/asyncstorage/sidecar.go#L80). 

### Additional context
This is a precursor to https://github.com/eclipse/che/issues/22308, see https://github.com/eclipse/che/issues/22308#issuecomment-1610069145

**Comment by l0rd** [comment]
Blocked by failing PR check tests that should be fixed as part of another issue.
